Bottom: 65b06b388fae5a13b47c67ef758394957a3c4c3f
Top:    ec9a508432c717f48ab24a7f448020a25ecde914
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-06 01:07:56 +0100

action.c: In ActionUnselect(), only hide the crosshair where required


---

diff --git a/src/action.c b/src/action.c
index 9621494..49ff7ac 100644
--- a/src/action.c
+++ b/src/action.c
@@ -5572,7 +5572,6 @@ ActionUnselect (int argc, char **argv, int x, int y)
   char *function = ARG (0);
   if (function)
     {
-      HideCrosshair ();
       switch (GetFunctionID (function))
 	{
 #if defined(HAVE_REGCOMP) || defined(HAVE_RE_COMP)
@@ -5630,6 +5629,7 @@ ActionUnselect (int argc, char **argv, int x, int y)
 			  Crosshair.AttachedBox.Point2.X);
 	    box.Y2 = MAX (Crosshair.AttachedBox.Point1.Y,
 			  Crosshair.AttachedBox.Point2.Y);
+	    HideCrosshair ();
 	    NotifyBlock ();
 	    if (Crosshair.AttachedBox.State == STATE_THIRD &&
 		SelectBlock (&box, false))
@@ -5637,6 +5637,7 @@ ActionUnselect (int argc, char **argv, int x, int y)
 		SetChangedFlag (true);
 		Crosshair.AttachedBox.State = STATE_FIRST;
 	      }
+	    RestoreCrosshair ();
 	    break;
 	  }
 
@@ -5665,12 +5666,10 @@ ActionUnselect (int argc, char **argv, int x, int y)
 	  break;
 
 	default:
-	  RestoreCrosshair ();
 	  AFAIL (unselect);
 	  break;
 
 	}
-      RestoreCrosshair ();
     }
   return 0;
 }
